src/store/create-invitation/saga.ts

```diff
@@ -33,16 +33,24 @@ export function* fetchInvite() {
   }
 }
 
-export function* saga() {
-  const channel = yield call(getAuthChannel);
-
+function* listenForUserLogin() {
+  const authChannel = yield call(getAuthChannel);
   while (true) {
-    const { userId = undefined } = yield take(channel, '*');
+    yield take(authChannel, Events.UserLogin);
+    yield spawn(fetchInvite);
+  }
+}
 
-    if (userId) {
-      yield spawn(fetchInvite);
-    } else {
-      yield put({ type: SagaActionTypes.Cancel });
-    }
+function* listenForUserLogout() {
+  const authChannel = yield call(getAuthChannel);
+  while (true) {
+    yield take(authChannel, Events.UserLogout);
+    yield put({ type: SagaActionTypes.Cancel });
   }
 }
+
+export function* saga() {
+  yield spawn(listenForUserLogin);
+  yield spawn(listenForUserLogout);
```

Comment:
Thought this would be cleaner. 

But I had 2 doubts

a) Are there any events besides `UserLogin` & `UserLogout` in the `authChannel`. Seems like previously we were just looking for any event "except" the login before cancel 🤔 

b) Do we even need to "listen" now? Since we're fetching everytime we're opening the invite dialogue.

Reply:
Yeah, at this point, you can just eliminate the auth event listeners. There were only 2 and this was written before we discovered you could filter based on the `type` so that's why it was listening for "*".

What we _could_ do is get rid of the custom Cancel event we have here and swap in the UserLogout event. Effectively line 10 was saying, "try to get the invite but a user logs out in the middle bail out." (based on the listener for user logout). But, I'm not sure it matters any more if we're going to fetch the data every time. If an invite fetch is still running when the user logs out... it doesn't really matter because the request has already been started. Maybe we just need to `reset` the data every time just before we start sending the api request.